### DISCUSSION TITLE
More info on system calls?

### DISCUSSION THREAD

**Discussion by jnoche** [mortbopet/Ripes#121] [Q&A] [answered]
I'm trying to use the file-related system calls, but the information in "Help>System calls" is a bit lacking.  For example, to use the "Open" system call, what do I put in the "flags" argument?

Do you have some more-detailed documentation about the system calls?  Or an example source file with file-related system calls?

**Comment by mortbopet** [comment, marked as answer]
The flags are intended to follow values generally used in a unix environment, but I agree that this is not really communicated in the system call dialog.

The flags are defined here:
https://github.com/mortbopet/Ripes/blob/d692d719e60dd9e6f043be337dbfa003a20dcba8/src/syscall/systemio.h#L80-L86
I'll add some issues to address this.

**Reply by jnoche** [reply]
Thanks for the very prompt response.  The code you provided answers my immediate question.  Thank you for providing the link; I'll use it to find answers in case I have questions in the future.

**Reply by matsievskiysv** [reply]
> The flags are intended to follow values generally used in a unix environment

FYI, values are different in Linux:

```c
#include <stdio.h>
#include <fcntl.h>

int
main(void)
{
	printf("O_RDONLY = 0x%x\n", O_RDONLY);
	printf("O_WRONLY = 0x%x\n", O_WRONLY);
	printf("O_RDWR = 0x%x\n", O_RDWR);
	printf("O_APPEND = 0x%x\n", O_APPEND);
	printf("O_CREAT = 0x%x\n", O_CREAT);
	printf("O_TRUNC = 0x%x\n", O_TRUNC);
	printf("O_EXCL = 0x%x\n", O_EXCL);
	return 0;
}
```
return
```
O_RDONLY = 0x0
O_WRONLY = 0x1
O_RDWR = 0x2
O_APPEND = 0x400
O_CREAT = 0x40
O_TRUNC = 0x200
O_EXCL = 0x80
```
